### DISCUSSION TITLE
Error: Cannot handle unknown node `footnoteReference`

### DISCUSSION THREAD

**Discussion by songololo** [syntax-tree/unist#40] [Q&A] [answered]
I am trying to use [`footnote`](https://github.com/syntax-tree/mdast#footnotes) types as defined in the `mdast` spec but encounter these types of error when the content is processed through `mdast-util-to-markdown`:

```
Error: Cannot handle unknown node `footnoteReference`
```

It seems that `mdast-util-to-markdown` only handles certain node types. Not sure if this is an issue where only certain nodes are supported, or if I have made a mistake in my tree.

```
{
  type: "root",
  children: [
    {
      type: "paragraph",
      children: [
        {
          type: "text",
          value: "Ref A:",
        },
        {
          type: "footnoteReference",
          identifier: "Wasserman1994",
          label: 1,
        },
      ],
      position: {
        start: {
          line: 1,
          column: 1,
          offset: 0,
        },
        end: {
          line: 1,
          column: 24,
          offset: 23,
        },
      },
    },
  ],
  position: {
    start: {
      line: 1,
      column: 1,
      offset: 0,
    },
    end: {
      line: 1,
      column: 24,
      offset: 23,
    },
  },
}
```

Any advice?

**Comment by wooorm** [comment, marked as answer]
Non-standard things are in separate projects. In this case, https://github.com/syntax-tree/mdast-util-footnote.

**Reply by songololo** [reply]
Thanks, I added and chained the `remark-footnotes` package and all works nicely now.
